src/plugins/auto_batch/src/plugin.cpp

```diff
@@ -49,311 +46,359 @@ DeviceInformation Plugin::ParseBatchDevice(const std::string& deviceWithBatch) {
         batch = std::stol(d.substr(openingBracket + 1, closingBracket - 1));
 
         if (batch <= 0) {
-            IE_THROW() << "Batch value for '" << deviceName << "' must be > 0, while " << batch << "is passed";
+            OPENVINO_THROW("Batch value for '", deviceName, "' must be > 0, while ", batch, "is passed");
         }
     }
-    return {deviceName, {{}}, batch};
+    return {deviceName, {{}}, static_cast<uint32_t>(batch)};
 }
 
-DeviceInformation Plugin::ParseMetaDevice(const std::string& devicesBatchCfg,
-                                          const std::map<std::string, std::string>& user_config) const {
-    auto metaDevice = ParseBatchDevice(devicesBatchCfg);
-    metaDevice.config = GetCore()->GetSupportedConfig(metaDevice.device_name, user_config);
-
+DeviceInformation Plugin::parse_meta_device(const std::string& devices_batch_config,
+                                            const ov::AnyMap& user_config) const {
+    auto meta_device = parse_batch_device(devices_batch_config);
+    meta_device.device_config = get_core()->get_supported_property(meta_device.device_name, user_config);
     // check that no irrelevant config-keys left
     for (const auto& k : user_config) {
         const auto& name = k.first;
-        if (metaDevice.config.find(name) == metaDevice.config.end() &&
+        if (meta_device.device_config.find(name) == meta_device.device_config.end() &&
             !ov::util::contains(supported_configKeys, name)) {
-            IE_THROW() << "Unsupported config key: " << name;
+            OPENVINO_THROW("Unsupported config key: ", name);
         }
     }
-    return metaDevice;
+    return meta_device;
 }
 
-InferenceEngine::RemoteContext::Ptr Plugin::CreateContext(const InferenceEngine::ParamMap& remote_properties) {
+std::shared_ptr<ov::IRemoteContext> Plugin::create_context(const ov::AnyMap& remote_properties) const {
     auto cfg = remote_properties;
+OPENVINO_SUPPRESS_DEPRECATED_START
     auto it = cfg.find(CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG));
+OPENVINO_SUPPRESS_DEPRECATED_END
     if (it == cfg.end())
         it = cfg.find(ov::device::priorities.name());
     if (it == cfg.end())
-        IE_THROW() << "Value for KEY_AUTO_BATCH_DEVICE_CONFIG is not set";
+        OPENVINO_THROW("Value for MULTI_DEVICE_PRIORITIES is not set");
 
     auto val = it->second.as<std::string>();
-    auto core = GetCore();
+    auto core = get_core();
     if (!core)
         return nullptr;
-    auto metaDevice = ParseMetaDevice(val, std::map<std::string, std::string>());
+    auto metaDevice = parse_meta_device(val, ov::AnyMap());
     cfg.erase(it);
-    return core->CreateContext(metaDevice.device_name, cfg);
+    auto remote_context = core->create_context(metaDevice.device_name, cfg);
+    return remote_context._impl;
```

Comment:
can we create a wrapper AutoBatchRemoteContext inherited from ov::IRemoteContext ? And return it here
In this case we don't need to extract internal implementation from public interface

Reply:
when will this api be called for batch? I mean either been explictly or implicitly called

Reply:
Agree with Ilya. You don't need to have a friend and you don't need to take impl (in this case you loose `so` object).
Please create AutoBatchTemoveContext. Use proxy plugin as an example: https://github.com/openvinotoolkit/openvino/pull/17416/files#diff-a4e0360e2a6f82e851e340c919c1f595e95fc53bcf86f2e776d555f8bc9718c9R15